src/Model_Hydro/CPU_Hydro/CPU_FluidSolver_MHM.cpp

```diff
@@ -543,10 +587,10 @@ void CPU_FluidSolver_MHM(
 
 //          1-a-5. evaluate the face-centered values by data reconstruction
 //                 --> note that g_PriVar_Half_1PG[] returned by Hydro_RiemannPredict() stores the primitive variables
-            Hydro_DataReconstruction( NULL, g_FC_Mag_Half_1PG, g_PriVar_Half_1PG, g_FC_Var_1PG, g_Slope_PPM_1PG,
+            Hydro_DataReconstruction( NULL, g_FC_Mag_Half_1PG, g_PriVar_Half_1PG, g_FC_Var_1PG, NULL, g_Slope_PPM_1PG,
```

Comment:
@ChunYen-Chen should `g_Flux` be `NULL` here?

Reply:
Yes, the half-step update is done by `Hydro_RiemannPredict()` for the `MHM_RP` scheme.